txnkv/transaction/commit.go

```diff
@@ -158,7 +159,17 @@ func (action actionCommit) handleSingleBatch(c *twoPhaseCommitter, bo *retry.Bac
 			if rejected := keyErr.GetCommitTsExpired(); rejected != nil {
 				logutil.Logger(bo.GetCtx()).Info("2PC commitTS rejected by TiKV, retry with a newer commitTS",
 					zap.Uint64("txnStartTS", c.startTS),
-					zap.Stringer("info", logutil.Hex(rejected)))
+					zap.Stringer("info", logutil.Hex(rejected)),
+				)
+
+				if !batch.isPrimary || !bytes.Equal(rejected.Key, c.primary()) {
+					logutil.Logger(bo.GetCtx()).Error("2PC commitTS rejected by TiKV, but the key is not the primary key",
+						zap.Uint64("txnStartTS", c.startTS),
+						zap.String("key", hex.EncodeToString(rejected.Key)),
```

Comment:
Do we have redact encode interfaces in client-go?

Reply:
No

Reply:
We may consider not logging the keys, so that we're not making the situation worse